R/fct_helpers.R

```diff
@@ -38,74 +54,320 @@ update_query <- function(active_table, column_name, value, rowid) {
-  res <- paste0("DELETE FROM ",
-                active_table)
+  res <- paste0('DELETE FROM "',
+                active_table, '"')
   return(res)
 }
 
 # Insert data into a table
 insert_query <- function(active_table, values) {
-  insert_query_values <- "("
+  insert_query_values <- '('
   i = 0
   for (val in values) {
     # Warning occurs here because of conversion into numeric type in if statement.
     i = i + 1
     if (!is.na(as.numeric(val))) {
       if (i != length(values))
         insert_query_values <-
-          paste0(insert_query_values, val, ",")
+          paste0(insert_query_values, val, ',')
       else
         insert_query_values <-
-          paste0(insert_query_values, val, ")")
+          paste0(insert_query_values, val, ')')
     }
     else{
       if (i != length(values))
         insert_query_values <-
-          paste0(insert_query_values, '"', val, '",')
+          paste0(insert_query_values, '\'', val, '\',')
       else
         insert_query_values <-
-          paste0(insert_query_values, '"', val, '")')
+          paste0(insert_query_values, '\'', val, '\')')
     }
   }
   res <-
-    paste0("INSERT INTO ",
+    paste0('INSERT INTO "',
            active_table,
-           " VALUES ",
+           '" VALUES ',
            insert_query_values)
   return(res)
-}
\ No newline at end of file
+}
+
+# Column definition query.
+
+column_details_query <- function(column_name = NULL,
+                                 data_type = NULL,
+                                 primary_key = NULL,
+                                 autoincrement_primary_key = NULL,
+                                 sort_order_primary_key = NULL,
+                                 on_conflict_primary_key = NULL,
+                                 unique = NULL,
+                                 on_conflict_unique = NULL,
+                                 not_null = NULL,
+                                 on_conflict_not_null = NULL,
+                                 default = NULL,
+                                 default_value_default = NULL,
+                                 check_condition = NULL,
+                                 specify_condition_check_condition = NULL,
+                                 collate = NULL,
+                                 collation_type_collate = NULL,
+                                 foreign_key = NULL,
+                                 foreign_table_foreign_key = NULL,
+                                 foreign_column_foreign_key = NULL,
+                                 on_update_foreign_key = NULL,
+                                 on_delete_foreign_key = NULL,
+                                 match_foreign_key = NULL,
+                                 defer_first_foreign_key = NULL,
+                                 defer_second_foreign_key = NULL) {
+  res <- paste0('"', column_name, '" "', data_type, '" ')
+  
+  if (isTRUE(primary_key)) {
+    res <- paste0(res, "PRIMARY KEY ")
```

Comment:
maybe `res <- sort_order_primary_key != "" ? paste0(res, sort_order_primary_key, " ") : ""` is shorter